modules/artifact-registry/iam.tf

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+moved {
+  from = google_artifact_registry_repository_iam_binding.bindings
+  to   = google_artifact_registry_repository_iam_binding.authoritative
+}
+
+resource "google_artifact_registry_repository_iam_binding" "authoritative" {
+  for_each   = var.iam
```

Comment:
I think you need google-beta here since the original block was using it

Reply:
Not sure we still need it as the commit introducing it was 4 years ago

Reply:
Yes, but the moved fails if the provider changes.

Reply:
Hm looks to be fine, maybe because the resource structure in `google` matches the one in `google-beta` 🤷 

```
  # module.docker_artifact_registry.google_artifact_registry_repository_iam_binding.bindings["roles/artifactregistry.reader"] has moved to module.docker_artifact_registry.google_artifact_registry_repository_iam_binding.authoritative["roles/artifactregistry.reader"]
    resource "google_artifact_registry_repository_iam_binding" "authoritative" {
        id         = "projects/xxxxx/locations/europe/repositories/public/roles/artifactregistry.reader"
        # (6 unchanged attributes hidden)
    }

  # module.docker_artifact_registry.google_artifact_registry_repository_iam_binding.bindings["roles/artifactregistry.repoAdmin"] has moved to module.docker_artifact_registry.google_artifact_registry_repository_iam_binding.authoritative["roles/artifactregistry.repoAdmin"]
    resource "google_artifact_registry_repository_iam_binding" "authoritative" {
        id         = "projects/xxxxx/locations/europe/repositories/public/roles/artifactregistry.repoAdmin"
        # (6 unchanged attributes hidden)
    }
```

Old state
```json
    {
      "module": "module.docker_artifact_registry",
      "mode": "managed",
      "type": "google_artifact_registry_repository_iam_binding",
      "name": "bindings",
      "provider": "provider[\"registry.terraform.io/hashicorp/google-beta\"]",
      "instances": [
```

New state 
```json
{
      "module": "module.docker_artifact_registry",
      "mode": "managed",
      "type": "google_artifact_registry_repository_iam_binding",
      "name": "authoritative",
      "provider": "provider[\"registry.terraform.io/hashicorp/google\"]",
      "instances": [
```

Reply:
Let me know if you prefer to keep that change for later

Reply:
This is what I get without the explitit `provider = google-beta`:

![image](https://github.com/user-attachments/assets/d5e3a517-811e-46e5-abea-a879cba65283)

Reply:
Let's go with this. If someone has issues we'll add it.